controller/src/main/java/io/pravega/controller/store/stream/StreamMetadataStore.java

```diff
@@ -1441,12 +1443,12 @@
      * @param context    operation context.
      * @param executor   callers executor.
      *                   
-     * @return A completable future which when completed will hold a boolean which will indicate if streamcut1 is strictly
-     * ahead of streamcut2. 
+     * @return A completable future which when completed will hold an integer which will determine the order between 
+     * streamcut1 and streamcut2. 
      */
-    CompletableFuture<Boolean> streamCutStrictlyGreaterThan(final String scope, final String streamName,
-                                                            Map<Long, Long> streamCut1, Map<Long, Long> streamCut2,
-                                                            final OperationContext context, final Executor executor);
+    CompletableFuture<Integer> compareStreamCut(final String scope, final String streamName,
```

Comment:
we could consider having a comparator for streamcuts.

Reply:
comparator's definition is 0 = equal. there is no value for overlap there. 
by using comparator interface we would be using it incorrectly for the definitions.